src/lib/profiles/data-management/Current/SubscriptionEngine.cpp

```diff
@@ -499,11 +499,22 @@ WEAVE_ERROR SubscriptionEngine::ProcessDataList(nl::Weave::TLV::TLVReader & aRea
         SchemaVersionRange versionRange;
 
         err = aCatalog->AddressToHandle(pathReader, handle, versionRange);
-        SuccessOrExit(err);
 
-        err = aCatalog->Locate(handle, &dataSink);
+        if (err == WEAVE_ERROR_INVALID_PROFILE_ID)
+        {
+            // Note: the application can remove sinks from the catalog
```

Comment:
nit: Rephrase as
"AddressToHandle() can return an error if the datasink for the handle has been removed from the catalog. In that case, continue to next entry".